lib/saluki-env/src/autodiscovery/providers/remote_agent.rs

```diff
@@ -0,0 +1,82 @@
+use std::time::Duration;
+
+use async_trait::async_trait;
+use futures::StreamExt;
+use tokio::sync::broadcast::{self, Receiver, Sender};
+use tokio::sync::OnceCell;
+use tracing::{debug, info, warn};
+
+use crate::autodiscovery::{AutodiscoveryEvent, AutodiscoveryProvider};
+use crate::helpers::remote_agent::RemoteAgentClient;
+
+/// An autodiscovery provider that uses the Datadog Agent's internal gRPC API to receive autodiscovery updates.
+pub struct RemoteAgentAutoDiscoveryProvider {
+    client: RemoteAgentClient,
+    sender: Sender<AutodiscoveryEvent>,
+    listener_init: OnceCell<()>,
+}
+
+impl RemoteAgentAutoDiscoveryProvider {
+    /// Creates a new `RemoteAgentAutoDiscoveryProvider` that uses the remote client to receive autodiscovery updates.
+    pub fn new(client: RemoteAgentClient) -> Self {
+        let (sender, _) = broadcast::channel::<AutodiscoveryEvent>(super::AD_STREAM_CAPACITY);
+
+        Self {
+            client,
+            sender,
+            listener_init: OnceCell::new(),
+        }
+    }
+
+    async fn start_background_listener(&self) {
+        debug!("Starting autodiscovery background listener.");
+
+        let mut client = self.client.clone();
+        let sender = self.sender.clone();
+
+        tokio::spawn(async move {
+            info!("Listening to autodiscovery events from remote agent.");
+
+            loop {
+                let mut autodiscovery_stream = client.get_autodiscovery_stream();
+
+                while let Some(result) = autodiscovery_stream.next().await {
```

Comment:
We might want to add a debug log right above this line to the effect of:

```rust
debug!("Polling autodiscovery event stream.");
```

While the info log we have above tells us that we've started listening _overall_, we don't have a way to know if the stream is ending gracefully and being restarted often, which adding this debug log would give us.